pkg/device42/paginate.go

```diff
@@ -0,0 +1,92 @@
+package device42
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"math"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// Paginator is an interface that provides a concurrent fan-out to retrieve multiple
```

Comment:
Question: do we want to make concurrent requests, given that we're creating this caching service partially to shield device42 from load?